src/main/java/org/blondin/mpg/Main.java

```diff
@@ -23,16 +30,38 @@ public static void main(String[] args) {
     }
 
     static void process(MpgClient mpgClient, MpgStatsClient mpgStatsClient) {
-
-        LOG.info("TODO: This is just a try for the moment ... display your players, stats available for {} players",
-                mpgStatsClient.getStats().getPlayers().size());
-
         for (League league : mpgClient.getDashboard().getLeagues()) {
             LOG.info("========== {} ==========", league.getName());
-            for (Player player : mpgClient.getCoach(league.getId()).getPlayers()) {
-                LOG.info("{}", player.getName());
+
+            List<Player> players = calculateEfficiency(mpgClient.getCoach(league.getId()).getPlayers(), mpgStatsClient);
+            Collections.sort(players, Comparator.comparing(Player::getPosition).thenComparing(Player::getEfficiency).reversed());
+            int nameMaxLength = players.stream().map(Player::getName).collect(Collectors.toList()).stream().max(Comparator.comparing(String::length))
+                    .orElse("").length();
+            Position lp = Position.G;
+            for (Player player : players) {
+
+                // Write position separator
+                if (!player.getPosition().equals(lp)) {
+                    lp = player.getPosition();
+                    String dashes = IntStream.range(0, nameMaxLength + 11).mapToObj(i -> "-").collect(Collectors.joining(""));
+                    LOG.info("{}", dashes);
+                }
+
+                // Write player
+                String spacesAfterName = IntStream.range(0, nameMaxLength - player.getName().length()).mapToObj(i -> " ")
+                        .collect(Collectors.joining(""));
+                LOG.info("{} | {}{} | {}", player.getPosition(), player.getName(), spacesAfterName, player.getEfficiency());
             }
         }
     }
 
+    static List<Player> calculateEfficiency(List<Player> players, MpgStatsClient stats) {
+        for (Player player : players) {
+            // TODO : First try with simple average
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/axel3rd-github/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.blondin%3Ampg-coach-bot&pullRequest=9&issues=AWZGqRCFnzfF1D4JAdMc&open=AWZGqRCFnzfF1D4JAdMc)

[ISSUE_KEY]: AWZGqRCFnzfF1D4JAdMc

Reply:
Will be fixed soon